netgear/netgear.go

```
@@ -0,0 +1,177 @@
+	resp, err := n.client.Get(logoutURL.String())
+	if err != nil {
+		return err
+	}
+	defer func() { _ = resp.Body.Close() }()
+	_, _ = io.Copy(io.Discard, resp.Body)
+	return nil
+}
+
+func (n *Netgear) DeviceInfo() (*DeviceInfo, error) {
+	var di DeviceInfo
+	if err := n.doRequest(http.MethodGet, "device_info", &di); err != nil {
+		return nil, err
+	}
+	return &di, nil
+}
+
+func (n *Netgear) PortStatistics(statType string) (*PortStatistics, error) {
+	const defaultPageIndex = 1
+	const defaultPageSize = 25
+
+	// NOTE: There should be a paginated request here!
+	// However, as of 2025-11-25, the API does not respect the parameters "indexPage" and "pageSize" and always
+	// returns the first page. Therefore, this request only works for switches with less than 25 ports.
+	u := n.baseUrl.JoinPath("port_statistics")
+	q := u.Query()
+	q.Set("type", statType)
+	q.Set("indexPage", strconv.Itoa(defaultPageIndex))
+	q.Set("pageSize", strconv.Itoa(defaultPageSize))
+	u.RawQuery = q.Encode()
+
+	var stats PortStatistics
+	if err := n.doRequestURL(http.MethodGet, u, &stats); err != nil {
+		return nil, err
+	}
+	return &stats, nil
+}
+
+func (n *Netgear) PoeStatus() (*PoeStatus, error) {
+	var poeStatus PoeStatus
+	if err := n.doRequest(http.MethodGet, "swcfg_poe", &poeStatus); err != nil {
+		return nil, err
+	}
+	return &poeStatus, nil
+}
+
+func (n *Netgear) doRequest(method, path string, result any) error {
+	return n.doRequestURL(method, n.baseUrl.JoinPath(path), result)
+}
+
+func (n *Netgear) doRequestURL(method string, u *url.URL, result any) error {
+	req, err := http.NewRequest(method, u.String(), nil)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("session", n.sessionToken)
+
+	resp, err := n.client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer func() { _ = resp.Body.Close() }()
+
+	if result == nil {
+		// ignore response body but still read it so connection can be reused
+		_, _ = io.Copy(io.Discard, resp.Body)
+		return nil
+	}
```

1. The body draining can be moved to the deferred function regardless of the result.
2. Why is a nil pointer for `result` valid? I have found no `doRequestURL` or `doRequest` method call with a nil pointer, indicating the result should be discarded. Thus, either let it crash or document why it is required.

---

2. While there is no place where we call `doRequestURL` without a result object, I still see value in allowing HTTP requests without requiring a json formatted body back.

---

Totally fine, but then it should be documented in the method doc comment. Otherwise, it would be unspecified/unexpected behavior.